proposals/infra/INF-0005-Long-Vector-TestPlan.md

```diff
@@ -0,0 +1,273 @@
+# Test Plan
+
+All tests are to be included in the HLK test binary which ships with the OS.
+This test binary is only built in the OS repo and based off of the
+ExecutionTests source code in the DXC repo. There is a script in the WinTools
+repo which generates and annotates the HLK tests.
+
+There are three test categories we are concerned with:
+
+1. Implement DXIL Intrinsic tests:
+    - At the bottom of this document there is a table of all HLSL intrinsics and
+    their mapped DXIL OpCodes. We just need coverage for each DXIL OpCode. If
+    the DXIL OpCode column lists 'emulated' then this means that there are
+    multiple LLVM/DXIL Ops used to compute the intrinsic. We will need to do an
+    audit of them to confirm coverage of all vector ops.
+
+2. Implement LLVM Native operation tests:
+     - These are the test cases for the 'basic' HLSL operators listed in the
+     HLSL operators table below.
+     - We have added some test cases for these with the long vectors work, but
+     we will want to make sure we add HLK coverage for all of these.
+
+3. Standard loading and storing of long vectors
+     - Ensure we have some basic tests doing standard loading/storing of long
+     vectors across buffer types. __TODO: Add details about which types of
+     buffers to test and why.__
+
+# Vector Sizes to test
+
+General sizes to test are in the range [5, 1024]. Sizes < 5 are assumed to
```

Comment:
The new vector load/store intrinisics produce vectorized versions for small vectors (<5) right? So those intrinsic tests will cover the <5 cases as well?

Reply:
Yes, sizes less than 5 are covered by existing test cases. I'll update to clarify that point.